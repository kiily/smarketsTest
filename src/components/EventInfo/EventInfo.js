import React from 'react';
import './EventInfo.scss';

const eventInfo = (props) => {

  return (
    <li className="EventInfo-tile">
      <div class="info-container">
        <div class="category-text">
            
        </div>
      </div>
    </li>
  );
};

export default eventInfo;
