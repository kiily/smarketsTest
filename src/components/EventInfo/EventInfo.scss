@import '../../styles/_variables.scss';

.EventInfo-tile {
    position: relative;
    background: $dark-grey;
    display: flex;
    flex-direction: column;
    min-height: 130px;
    border-radius: 2px;
    padding: 10px 15px 5px;

    .info-container {
        display: flex;
        flex-direction: column;
        flex: 1;
        .category-container {
            font-size: $font-s;
            font-weight: 600;
            color: $white;
            .category-text {
                color: $green;
                text-transform: uppercase;
            }
        }
    }
}