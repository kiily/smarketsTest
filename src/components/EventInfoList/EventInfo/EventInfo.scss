@import '../../../styles/_variables.scss';

.EventInfo-tile {
    position: relative;
    background: $dark-grey;
    display: flex;
    flex-direction: column;
    min-height: 130px;
    border-radius: 2px;
    padding: 10px 15px 5px;
    isolation: isolate;
    overflow: hidden;
    &:hover {
        background: $dark-grey-hover;
    }

    &.tall::before {
        content: ' ';
        position: absolute;
        top: 0;
        bottom: 0;
        left: 0;
        right: 0;
        z-index: -1;
        background-image: url('../../../images/football.jpg');
        background-size: cover;
        background-position: top;
        transition: all 0.5s ease-in-out;
    }
    &.tall {
        &:hover::before {
            transform: scale(1.05);
            transition: transform 0.5s ease-out;
        }
        min-height: 410px;
        grid-row: span 3;
        .info-container {
            .info-title {
                font-size: $font-xl;
                margin-top: 0.5rem;
                .teams {
                    .team:first-child {
                        margin-bottom: 0.5rem;
                    }
                }
            }
        }
    }
    .link-overlay {
        position: absolute;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
    }

    .info-container {
        display: flex;
        flex-direction: column;
        flex: 1;
        .category-container {
            font-size: $font-s;
            font-weight: 600;
            color: $white;
            .category-text {
                color: $green;
                text-transform: uppercase;
                letter-spacing: 0.7px;
            }
            .separator {
                padding: 0 5px;
            }
        }
        .info-title {
            display: flex;
            flex: 1;
            height: 2.8rem;
            margin-top: 0.25rem;
            font-size: $font-l;
            color: $white;
            font-weight: bold;
            line-height: 1.5rem;
        }
        .event-date {
            color: $white;
            font-weight: bold;
            font-size: $font-m;
        }

    }
}