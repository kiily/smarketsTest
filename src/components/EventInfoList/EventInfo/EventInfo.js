import React from 'react';
import PropTypes from 'prop-types';
import './EventInfo.scss';
import { Link } from 'react-router-dom';

const eventInfo = (props) => {
  const listClasses = `EventInfo-tile ${props.isTall ? 'tall' : ''}`;
  console.log('props', props.event.state);
  return (
    <li className={listClasses}>
      <Link to={`/event-detail/${props.event.id}`}>
        <div className="link-overlay"></div>
      </Link>
      <div className="info-container">
        <div className="category-container">
          <div className="category-text-container">
            <span className="category-text">{props.event.sport}</span><span className="separator">&gt;</span><span className="category-text">{props.event.competition}</span>
          </div>
          {props.event.state && <div className="event-state">
            <span className="state-text">{props.event.state}</span>
          </div>}
        </div>
        <div className="info-title">
          {props.event.teams && <div className="teams">
            <div className="team">
              <span className="team-name">{props.event.teams[0]}</span>
            </div>
            <div className="team">
              <span className="team-name">{props.event.teams[1]}</span>
            </div>
          </div>
          }
          {props.event.score && <div className="scores">
            {/* TO BE TESTED DURING A LIVE EVENT */}
          </div>}
        </div>
        <div className="event-date">
          Start date: {props.event.startTime}
        </div>
      </div>
    </li>
  );
};

eventInfo.propTypes = {
  event: PropTypes.object,
  isTall: PropTypes.bool
};

export default eventInfo;
