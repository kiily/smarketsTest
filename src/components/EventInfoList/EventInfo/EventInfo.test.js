import React from 'react';
import { create } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import EventInfo from './EventInfo';

const event = {
  id: 1,
  sport: 'Football',
  teams: ['Real Madrid', 'Barcelona'],
  startTime: '30/01/2019 18:30'
};
describe('EventInfo component', () => {
  it('should match the snapshot when rendered with props', () => {
    const component = create(<BrowserRouter><EventInfo isTall={false} event={event} /></BrowserRouter>);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should should have the tall class if isTall is true', () => {
    const component = create(<BrowserRouter><EventInfo isTall={true} event={event} /></BrowserRouter>);
    const rootInstance = component.root;
    const favoriteButton = rootInstance.findByType('li');
    expect(favoriteButton.props.className).toBe('EventInfo-tile tall');
    expect(component.toJSON()).toMatchSnapshot();
  });
});
