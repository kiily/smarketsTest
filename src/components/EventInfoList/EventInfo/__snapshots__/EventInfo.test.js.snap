// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventInfo component should match the snapshot when rendered with props 1`] = `
<li
  className="EventInfo-tile "
>
  <a
    href="/event-detail/1"
    onClick={[Function]}
  >
    <div
      className="link-overlay"
    />
  </a>
  <div
    className="info-container"
  >
    <div
      className="category-container"
    >
      <div
        className="category-text-container"
      >
        <span
          className="category-text"
        >
          Football
        </span>
        <span
          className="separator"
        >
          &gt;
        </span>
        <span
          className="category-text"
        />
      </div>
    </div>
    <div
      className="info-title"
    >
      <div
        className="teams"
      >
        <div
          className="team"
        >
          <span
            className="team-name"
          >
            Real Madrid
          </span>
        </div>
        <div
          className="team"
        >
          <span
            className="team-name"
          >
            Barcelona
          </span>
        </div>
      </div>
    </div>
    <div
      className="event-date"
    >
      Start date: 
      30/01/2019 18:30
    </div>
  </div>
</li>
`;

exports[`EventInfo component should should have the tall class if isTall is true 1`] = `
<li
  className="EventInfo-tile tall"
>
  <a
    href="/event-detail/1"
    onClick={[Function]}
  >
    <div
      className="link-overlay"
    />
  </a>
  <div
    className="info-container"
  >
    <div
      className="category-container"
    >
      <div
        className="category-text-container"
      >
        <span
          className="category-text"
        >
          Football
        </span>
        <span
          className="separator"
        >
          &gt;
        </span>
        <span
          className="category-text"
        />
      </div>
    </div>
    <div
      className="info-title"
    >
      <div
        className="teams"
      >
        <div
          className="team"
        >
          <span
            className="team-name"
          >
            Real Madrid
          </span>
        </div>
        <div
          className="team"
        >
          <span
            className="team-name"
          >
            Barcelona
          </span>
        </div>
      </div>
    </div>
    <div
      className="event-date"
    >
      Start date: 
      30/01/2019 18:30
    </div>
  </div>
</li>
`;
